src/OAuth2/Encryption/Jwt.php

```diff
@@ -142,6 +145,25 @@ public function urlSafeB64Decode($b64)
         return base64_decode($b64);
     }
 
+    public function getSupportedAlgorithms($type = null)
+    {
+        if ($type === null) {
+            return array_keys(self::$supportedAlgorithms);
+        }
+        else {
+            $filtered = array();
+            foreach (self::$supportedAlgorithms as $alg => $method) {
+                if ($type === 'RSA' && $alg[0] === 'R') {
```

Comment:
Yeah I wasn't exactly thrilled with the implementation either, but it seemed like the quickest way.